src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -195,6 +197,60 @@ public TutorialClass findTutorialClassFromList(TutorialClass tutorialClass, Modu
         return tutorialClassInList;
     }
 
+    /**
+     * Randomly allocates the students in {@code tutorial class} into {@code numOfTeams} of different teams.
+     *
+     * @param moduleCode of the tutorial class.
+     * @param tutorialClass to allocate the different students into the teams to.
+     * @param numOfTeams of teams to split into.
+     */
+    public void randomTeamAllocation(ModuleCode moduleCode, TutorialClass tutorialClass, int numOfTeams) {
+        requireAllNonNull(moduleCode, tutorialClass, numOfTeams);
+        ArrayList<TutorialTeam> teams = tutorialClass.getTeams();
+        ArrayList<Person> classList = tutorialClass.getStudents();
+        int classSize = tutorialClass.getStudents().size();
+        int teamSize = (int) Math.ceil((double) classSize / numOfTeams);
+        teams.clear();
+
+        // creating the teams to add into
+        for (int i = 1; i <= numOfTeams; i++) {
+            String teamName = "Team" + i;
+            TutorialTeam team = new TutorialTeam(teamName, teamSize);
+            addTeam(tutorialClass, team);
+        }
+
+        Random random = new Random();
+        int k = 0;
+        while (k < classList.size()) {
```

Comment:
Also seems like you're assessing classList.size() instead of using the previously declared variable of classSize

Reply:
Corrected this to use the variable defined at the start of the function. Thanks!